init/eessi_environment_variables

```diff
@@ -0,0 +1,67 @@
+# Allow for a silent mode
+if [[ -v EESSI_SILENT ]]; then
+  # EESSI_SILENT set
+  output=/dev/null
+else
+  output=/dev/stdout
+fi
+
+export EESSI_PILOT_VERSION="2020.10"
+export EESSI_PREFIX=/cvmfs/pilot.eessi-hpc.org/$EESSI_PILOT_VERSION
+
+if [ -d $EESSI_PREFIX ]; then
+  echo "Found EESSI pilot repo @ $EESSI_PREFIX!" >> $output
+
+  # Set EPREFIX since that is basically a standard in Gentoo Prefix
+  export EPREFIX=$EESSI_PREFIX/compat/$(uname -m)
+  export EESSI_EPREFIX=$EPREFIX
+  if [ -d $EESSI_EPREFIX ]; then
+
+    # determine subdirectory in software layer
+    # note: eessi_software_subdir_for_host.py will pick up value from $EESSI_SOFTWARE_SUBDIR_OVERRIDE if it's defined!
+    export EESSI_EPREFIX_PYTHON=$EESSI_EPREFIX/usr/bin/python3
+    export EESSI_SOFTWARE_SUBDIR=$($EESSI_EPREFIX_PYTHON ${EESSI_PREFIX}/init/eessi_software_subdir_for_host.py $EESSI_PREFIX)
+    if [ ! -z $EESSI_SOFTWARE_SUBDIR ]; then
+
+        echo "Using ${EESSI_SOFTWARE_SUBDIR} as software subdirectory." >> $output
+        export EESSI_SOFTWARE_PATH=$EESSI_PREFIX/software/$EESSI_SOFTWARE_SUBDIR
+        if [ -d $EESSI_SOFTWARE_PATH ]; then
+          # Allow for the use of a custom MNS
+          if [ -z ${EESSI_CUSTOM_MODULEPATH+x} ]; then
+            # EESSI_CUSTOM_MODULEPATH not set so we use our defaults
+
+            # Allow for use of alternative module tree shipped with EESSI
+            if [ -z ${EESSI_MODULE_SUBDIR+x} ]; then
+              # EESSI_MODULE_SUBDIR not set
+              EESSI_MODULE_SUBDIR="modules/all"
+            fi
+            EESSI_MODULEPATH=$EESSI_SOFTWARE_PATH/$EESSI_MODULE_SUBDIR
```

Comment:
This is a way to support not just alternative MNS, but also domain-specific module trees (I can imagine combining this with domain easystack files)